File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/java/dmn/Decision_014_1.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"decision.ftl", "decision_014_1"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "decision_014_1",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class Decision_014_1 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision<com.gs.dmn.runtime.Context> {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "decision_014_1",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION,
        com.gs.dmn.runtime.annotation.ExpressionKind.CONTEXT,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private final Decision_014_3 decision_014_3;

    public Decision_014_1() {
        this(new Decision_014_3());
    }

    public Decision_014_1(Decision_014_3 decision_014_3) {
        this.decision_014_3 = decision_014_3;
    }

    @java.lang.Override()
    public com.gs.dmn.runtime.Context applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(input_.get("inputData_014_1"), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'Decision_014_1'", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.Context apply(String inputData_014_1, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start decision 'decision_014_1'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long decision_014_1StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments decision_014_1Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            decision_014_1Arguments_.put("inputData_014_1", inputData_014_1);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, decision_014_1Arguments_);

            // Evaluate decision 'decision_014_1'
            com.gs.dmn.runtime.Context output_ = lambda.apply(inputData_014_1, context_);

            // End decision 'decision_014_1'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, decision_014_1Arguments_, output_, (System.currentTimeMillis() - decision_014_1StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'decision_014_1' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<com.gs.dmn.runtime.Context> lambda =
        new com.gs.dmn.runtime.LambdaExpression<com.gs.dmn.runtime.Context>() {
            public com.gs.dmn.runtime.Context apply(Object... args_) {
                String inputData_014_1 = 0 < args_.length ? (String) args_[0] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 1 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[1] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                String decision_014_3 = Decision_014_1.this.decision_014_3.apply(context_);

                String decisionService_014 = DecisionService_014.instance().apply("A", "B", context_);
                com.gs.dmn.runtime.Context decision_014_1 = new com.gs.dmn.runtime.Context();
                decision_014_1.put("inputData_014_1",  inputData_014_1);
                decision_014_1.put("decision_014_3",  decision_014_3);
                decision_014_1.put("decisionService_014",  decisionService_014);
                return decision_014_1;
            }
        };
}

File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/python/dmn/Decision_014_1.py
import typing
import decimal
import datetime
import isodate
import time

import jdmn.runtime.Context
import jdmn.runtime.DefaultDMNBaseDecision
import jdmn.runtime.ExecutionContext
import jdmn.runtime.LambdaExpression
import jdmn.runtime.LazyEval
import jdmn.runtime.Pair
import jdmn.runtime.Range
import jdmn.runtime.RuleOutput
import jdmn.runtime.RuleOutputList

import jdmn.runtime.annotation.Annotation
import jdmn.runtime.annotation.AnnotationSet
import jdmn.runtime.annotation.DRGElementKind
import jdmn.runtime.annotation.ExpressionKind
import jdmn.runtime.annotation.HitPolicy

import jdmn.runtime.cache.Cache

import jdmn.runtime.external.ExternalFunctionExecutor

import jdmn.runtime.listener.Arguments
import jdmn.runtime.listener.DRGElement
import jdmn.runtime.listener.EventListener
import jdmn.runtime.listener.Rule

import Decision_014_3

import DecisionService_014


# Generated(value = ["decision.ftl", "decision_014_1"])
class Decision_014_1(jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision):
    DRG_ELEMENT_METADATA: jdmn.runtime.listener.DRGElement.DRGElement = jdmn.runtime.listener.DRGElement.DRGElement(
        "http://www.montera.com.au/spec/DMN/0085-decision-services",
        "decision_014_1",
        "",
        jdmn.runtime.annotation.DRGElementKind.DRGElementKind.DECISION,
        jdmn.runtime.annotation.ExpressionKind.ExpressionKind.CONTEXT,
        jdmn.runtime.annotation.HitPolicy.HitPolicy.UNKNOWN,
        -1
    )

    def __init__(self, decision_014_3: Decision_014_3.Decision_014_3 = None):
        jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision.__init__(self)
        self.decision_014_3 = Decision_014_3.Decision_014_3() if decision_014_3 is None else decision_014_3

    def apply(self, inputData_014_1: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[jdmn.runtime.Context.Context]:
        try:
            # Start decision 'decision_014_1'
            annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet = None if context_ is None else context_.annotations
            eventListener_: jdmn.runtime.listener.EventListener.EventListener = None if context_ is None else context_.eventListener
            externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor = None if context_ is None else context_.externalFunctionExecutor
            cache_: jdmn.runtime.cache.Cache.Cache = None if context_ is None else context_.cache
            decision_014_1StartTime_ = int(time.time_ns()/1000)
            decision_014_1Arguments_ = jdmn.runtime.listener.Arguments.Arguments()
            decision_014_1Arguments_.put("inputData_014_1", inputData_014_1)
            eventListener_.startDRGElement(self.DRG_ELEMENT_METADATA, decision_014_1Arguments_)

            # Evaluate decision 'decision_014_1'
            output_: typing.Optional[jdmn.runtime.Context.Context] = self.evaluate(inputData_014_1, context_)

            # End decision 'decision_014_1'
            eventListener_.endDRGElement(self.DRG_ELEMENT_METADATA, decision_014_1Arguments_, output_, (int(time.time_ns()/1000) - decision_014_1StartTime_))

            return output_
        except Exception as e:
            self.logError("Exception caught in 'decision_014_1' evaluation", e)
            return None

    def evaluate(self, inputData_014_1: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[jdmn.runtime.Context.Context]:
        # Apply child decisions
        decision_014_3: typing.Optional[str] = self.decision_014_3.apply(context_)

        decisionService_014: typing.Optional[str] = DecisionService_014.DecisionService_014.instance().apply("A", "B", context_)
        decision_014_1: typing.Optional[jdmn.runtime.Context.Context] = jdmn.runtime.Context.Context()
        decision_014_1.add("inputData_014_1",  inputData_014_1)
        decision_014_1.add("decision_014_3",  decision_014_3)
        decision_014_1.add("decisionService_014",  decisionService_014)
        return decision_014_1
